CL/cl_ext.h

```diff
@@ -225,6 +225,38 @@ typedef CL_API_ENTRY cl_command_queue
 #define CL_DEVICE_GPU_OVERLAP_NV                    0x4004
 #define CL_DEVICE_KERNEL_EXEC_TIMEOUT_NV            0x4005
 #define CL_DEVICE_INTEGRATED_MEMORY_NV              0x4006
+#define CL_DEVICE_ATTRIBUTE_ASYNC_ENGINE_COUNT_NV   0x4007
+#define CL_DEVICE_PCI_BUS_ID_NV                     0x4008
+#define CL_DEVICE_PCI_SLOT_ID_NV                    0x4009
+#define CL_DEVICE_PCI_DOMAIN_ID_NV                  0x400A
+#define CL_DEVICE_MAX_LOCAL_MEMORY_PER_SM_NV        0x400B
+
+/******************************************
+* cl_nv_create_buffer extension           *
+******************************************/
+typedef cl_bitfield         cl_mem_flags_NV;
+
+CL_API_ENTRY cl_mem CL_API_CALL
+clCreateBufferNV(cl_context     context,
+                 cl_mem_flags     flags,
+                 cl_mem_flags_NV  flags_NV,
+                 size_t           size,
+                 void             *host_ptr,
+                 cl_int           *errcode_ret) CL_EXT_SUFFIX__VERSION_1_2;
+
+
+#define CL_MEM_LOCATION_HOST_NV                     (1 << 0)
+#define CL_MEM_PINNED_NV                            (1 << 1)
+
+
+/*********************************
```

Comment:
Here's an AMD extension that was added in this PR that should be added in #119 instead.

Reply:
CL_MEM_PINNED_NV  is a nvidia define.

Reply:
Sorry, this comment applies to the code block below, for `cl_amd_device_memory_flags`.

Reply:
I can see more attribute queries after CL_DEVICE_MAX_LOCAL_MEMORY_PER_SM_NV:
```
#define CL_DEVICE_MAX_LOCAL_MEMORY_PER_SM_NV        0x400B
#define CL_DEVICE_UUID_KHR                          0x106A
#define CL_DRIVER_UUID_KHR                          0x106B
#define CL_DEVICE_LUID_VALID_KHR                    0x106C
#define CL_DEVICE_LUID_KHR                          0x106D
#define CL_DEVICE_NODE_MASK_KHR                     0x106E
#define CL_UUID_SIZE_KHR                            16
#define CL_LUID_SIZE_KHR                            8
```

Reply:
Also I see (which seems not included in these patches):
```
/******************************************
* cl_set_kernel_attribute_nv extension *
*******************************************/
typedef enum kernel_attribute_enum {
    CL_KERNEL_PREFERRED_LOCAL_MEMORY_SIZE_NV = 0,   /* setting preferred shared memory size */
}cl_kernel_attribute_nv;

CL_API_ENTRY cl_int CL_API_CALL
clSetKernelAttributeNV(cl_kernel kernel,
                       cl_device_id device,
                       cl_kernel_attribute_nv k_attr,
                       size_t param_value_size,
                       const void *param_value);

CL_API_ENTRY cl_int CL_API_CALL
clGetKernelAttributeNV(cl_kernel kernel,
                       cl_device_id device,
                       cl_kernel_attribute_nv k_attr,
                       void *param_value,
                       size_t *param_value_size_ret);
```

Reply:
the last one seems an upcoming extensions as NV 455.32 drv binary has "cl_kernel_attribute_nv".. a strange name for an extension..

Reply:
The name is being fixed in the next release to cl_nv_kernel_attribute as per extension naming convention.

Reply:
> 
> 
> I can see more attribute queries after CL_DEVICE_MAX_LOCAL_MEMORY_PER_SM_NV:
> 
> ```
> #define CL_DEVICE_MAX_LOCAL_MEMORY_PER_SM_NV        0x400B
> #define CL_DEVICE_UUID_KHR                          0x106A
> #define CL_DRIVER_UUID_KHR                          0x106B
> #define CL_DEVICE_LUID_VALID_KHR                    0x106C
> #define CL_DEVICE_LUID_KHR                          0x106D
> #define CL_DEVICE_NODE_MASK_KHR                     0x106E
> #define CL_UUID_SIZE_KHR                            16
> #define CL_LUID_SIZE_KHR                            8
> ```

These KHR enums are already be part of cl_ext.h (https://github.com/KhronosGroup/OpenCL-Headers/blob/9fac4e9866a961f66bdd72fa2bff50145512f972/CL/cl_ext.h#L644).